halo/attest/keeper/common_test.go

```diff
@@ -0,0 +1,62 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/omni-network/omni/halo/attest/keeper"
+	"github.com/omni-network/omni/halo/attest/testutil"
+	"github.com/omni-network/omni/halo/attest/types"
+
+	"cosmossdk.io/core/store"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	ctestutil "github.com/cosmos/cosmos-sdk/testutil"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+type mocks struct {
+	// store   *testutil.KVStoreService
+	cdc     *testutil.MockBinaryCodec
+	skeeper *testutil.MockStakingKeeper
+	voter   *testutil.MockVoter
+	namer   *testutil.MockChainNamer
+}
+
+func mockDefaultExpectations(_ sdk.Context, m mocks) {
+}
+
+func setupKeeper(t *testing.T, expectations ...func(sdk.Context, mocks)) (*keeper.Keeper, store.KVStoreService, sdk.Context) {
+	t.Helper()
+
+	key := storetypes.NewKVStoreKey(types.StoreKey)
+	storeSvc := runtime.NewKVStoreService(key)
+	// testCtx := ctestutil.DefaultContextWithDB(t, key, storetypes.NewTransientStoreKey("test_key"))
+	// ctx := testCtx.Ctx.WithHeaderInfo(header.Info{Time: time.Now()})
+	ctx := ctestutil.DefaultContext(key, storetypes.NewTransientStoreKey("test_key"))
+
+	// gomock initialization
+	ctrl := gomock.NewController(t)
+	m := mocks{
+		cdc:     testutil.NewMockBinaryCodec(ctrl),
+		skeeper: testutil.NewMockStakingKeeper(ctrl),
+		voter:   testutil.NewMockVoter(ctrl),
+		namer:   testutil.NewMockChainNamer(ctrl),
+	}
+	if len(expectations) == 0 {
+		mockDefaultExpectations(ctx, m)
+	} else {
+		for _, exp := range expectations {
+			exp(ctx, m)
+		}
+	}
+
+	k, err := keeper.New(m.cdc, storeSvc, m.skeeper, m.voter, m.namer.ChainName)
+	require.NoError(t, err, "error ")
+	if err != nil {
+		t.Fatalf("error creating keeper: %v", err)
+	}
```

Comment:
no need to handle errors twice in testing. The `require.NoError` above is sufficient.